samples/nextjs/src/lib/tracking-service.ts

```diff
@@ -1,7 +1,43 @@
-import { TrackingService } from '@sitecore-jss/sitecore-jss-tracking';
+import { LayoutServiceContext, RouteData } from '@sitecore-jss/sitecore-jss-nextjs';
+import {
+  PageViewData,
+  TrackingService as SitecoreTrackingService,
+  TrackingServiceConfig,
+} from '@sitecore-jss/sitecore-jss-tracking';
 import config from 'temp/config';
 
-export const trackingService = new TrackingService({
-  endpoint: config.trackingEndpoint,
-  siteName: config.jssAppName,
-});
+export class TrackingService extends SitecoreTrackingService {
```

Comment:
Can you please explain why this extension is needed in the sample? (as opposed to updating the base class)

Reply:
We want to have global isTrackignEnabled flag on the sample level and use it to control whether tracking is enabled or not. Instead of writing if(isTrackignEnabled) in different places we decided to create separate class which incapsulate this logic on the sample level. Adding this logic to the base class will look strange as TrackingService which takes isTrackingEnabled config param sounds wired.